include/dlaf/eigensolver/reduction_to_band/mc.h

```diff
@@ -307,6 +323,223 @@ void trmmComputeW(PanelT<Coord::Col, T>& w, MatrixLikeT& v, hpx::shared_future<C
     hpx::dataflow(ex, trmm_func, w(index_i), v.read(index_i), tile_t);
 }
 
+template <class T, class MatrixLikeT>
+void gemmUpdateX(PanelT<Coord::Col, T>& x, ConstMatrixT<T>& w2, MatrixLikeT& v) {
+  using matrix::unwrapExtendTiles;
+  using tile::gemm_o;
+
+  const auto ex = getHpExecutor<Backend::MC>();
+
+  // GEMM X = X - 0.5 . V . W2
+  for (const auto& index_i : v.iteratorLocal())
+    hpx::dataflow(ex, unwrapExtendTiles(gemm_o), blas::Op::NoTrans, blas::Op::NoTrans, T(-0.5),
+                  v.read(index_i), w2.read(LocalTileIndex(0, 0)), T(1), x(index_i));
+}
+
+template <class T>
+void hemmComputeX(PanelT<Coord::Col, T>& x, const LocalTileSize at_offset, ConstMatrixT<T>& a,
```

Comment:
The implementation of `hemmComputeX` `her2kUpdateTrailingMatrix` are different for local and distributed version. This is mainly because the distributed version relies on "transposed" support panels (e.g. `xt` for `x`), while the local version has everything locally and it does not need it.

Before spending time on this problem, I'd like to hear from you what you think and/or if you have any suggestion on how to address it.

Reply:
Is the question whether it's acceptable that the implementations are separate or not? Or is there something more to this? Is the local version a pure optimization or does it require a different implementation? In my opinion keeping them separate is perfectly fine, especially if they're optimized separately for the local/distributed cases. I suppose the `dataflow` calls could be separated into helper functions as we do elsewhere?

Reply:
As discussed during the meeting it is fine to have different implementations, anyway as @msimberg suggest I would use helper functions for dataflow.